src/test/java/com/stripe/net/ApiServiceTest.java

```diff
@@ -0,0 +1,176 @@
+package com.stripe.net;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.stripe.Stripe;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.exception.StripeException;
+import com.stripe.model.HasId;
+import com.stripe.model.StripeCollection;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.mockito.Mockito;
+
+class ApiServiceTest {
+  private static final String FOO_ID = "foo_123";
+  private static final String BAR_ID = "bar_123";
+
+  private static class Foo extends ApiResource implements HasId {
+    @Override
+    public String getId() {
+      return FOO_ID;
+    }
+  }
+
+  private static class FooCollection extends StripeCollection<Foo> {
+  }
+
+
+  private static class FooParams extends ApiRequestParams {
+  }
+
+  private static class FooOnBarParams extends ApiRequestParams {
+  }
+
+  private static class FooService extends ApiService {
```

Comment:
Were an omit and retrieve omitted here intentionally (I suspect that yes because they're trivial on top of the other method types, but just wanted to make sure).

Reply:
I wasn't as intentional about testing all methods, but wanted to cover both request(POST/DELETE) and request collection(GET).

I've added more methods still to cover the 5 standard methods plus the custom one.